diff --git a/src/mlpack/CMakeLists.txt b/src/mlpack/CMakeLists.txt
index bf9f3886d..70b767b6a 100644
--- a/src/mlpack/CMakeLists.txt
+++ b/src/mlpack/CMakeLists.txt
@@ -17,6 +17,15 @@ foreach(dir ${DIRS})
   add_subdirectory(${dir})
 endforeach()
 
+find_package(Boost
+        COMPONENTS
+        program_options
+        serialization
+        REQUIRED)
+
+target_link_libraries(mlpack ${Boost_LIBRARIES})
+target_include_directories(mlpack PUBLIC ${Boost_INCLUDE_DIRS})
+
 # MLPACK_SRCS is set in the subdirectories.  The dependencies (MLPACK_LIBRARIES)
 # are set in the root CMakeLists.txt.
 add_library(mlpack ${MLPACK_SRCS})
