diff --git a/src/mlpack/CMakeLists.txt b/src/mlpack/CMakeLists.txt
index bf9f3886d..ac7f369ca 100644
--- a/src/mlpack/CMakeLists.txt
+++ b/src/mlpack/CMakeLists.txt
@@ -56,7 +56,16 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   endif()
 endif()
 
+find_package(Boost
+        COMPONENTS
+        program_options
+        serialization
+        REQUIRED)
+
 set_target_properties(mlpack PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "prereqs.hpp")
+target_link_libraries(mlpack ${Boost_LIBRARIES})
+target_include_directories(mlack ${Boost_INCLUDE_DIRS})
+
 cotire(mlpack)
 
 if (BUILD_TESTS)
