From 4b32aea75c34830c67b4280a7e526b30463dc27a Mon Sep 17 00:00:00 2001
From: Ryan Curtin <ryan@ratml.org>
Date: Wed, 2 Nov 2022 11:42:37 -0400
Subject: [PATCH] Don't specify --prefix on Windows.

---
 src/mlpack/bindings/python/PythonInstall.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/mlpack/bindings/python/PythonInstall.cmake b/src/mlpack/bindings/python/PythonInstall.cmake
index 6e25fb926e..9cc2b02012 100644
--- a/src/mlpack/bindings/python/PythonInstall.cmake
+++ b/src/mlpack/bindings/python/PythonInstall.cmake
@@ -8,6 +8,11 @@ if (DEFINED ENV{DESTDIR})
           --prefix=${PYTHON_INSTALL_PREFIX} --root=$ENV{DESTDIR}
       WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/src/mlpack/bindings/python/"
       RESULT_VARIABLE setup_res)
+elseif (WIN32)
+  execute_process(COMMAND ${PYTHON_EXECUTABLE}
+      "${CMAKE_BINARY_DIR}/src/mlpack/bindings/python/setup.py" install
+      WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/src/mlpack/bindings/python/"
+      RESULT_VARIABLE setup_res)
 else ()
   execute_process(COMMAND ${PYTHON_EXECUTABLE}
       "${CMAKE_BINARY_DIR}/src/mlpack/bindings/python/setup.py" install
-- 
2.37.2

