diff --git a/CMakeLists.txt b/CMakeLists.txt
index 26757a5..5d4093d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,8 @@ option(MATHJAX
 option(FORCE_CXX11
     "Don't check that the compiler supports C++11, just assume it.  Make sure to specify any necessary flag to enable C++11 as part of CXXFLAGS." OFF)
 option(USE_OPENMP "If available, use OpenMP for parallelization." ON)
+add_definitions( -DBOOST_ALL_NO_LIB )
 enable_testing()
 
 # Set required standard to C++11.
